spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/observation/web/servlet/WebMvcObservationAutoConfiguration.java

```diff
@@ -97,4 +108,39 @@ MeterFilter metricsHttpServerUriTagFilter(ObservationProperties observationPrope
 
 	}
 
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnProperty(value = "management.observations.http.server.actuator.enabled", havingValue = "false")
+	static class ActuatorWebEndpointObservationConfiguration {
+
+		@Bean
+		ObservationPredicate actuatorWebEndpointObservationPredicate(ServerProperties serverProperties,
+				WebMvcProperties webMvcProperties, PathMappedEndpoints pathMappedEndpoints) {
+			return (name, context) -> {
+				if (context instanceof ServerRequestObservationContext serverContext) {
+					String endpointPath = getEndpointPath(serverProperties, webMvcProperties, pathMappedEndpoints);
```

Comment:
Since endpoint path doesn't change in runtime, I think it doesn't need to be resolved on each invocation of the predicate but rather only once when the predicate is set up.